src/io.rs

```diff
@@ -0,0 +1,198 @@
+use crate::units::preamble::*;
+
+use crate::network::RoadNetwork;
+use crate::road_functions::IntersectionType;
+use crate::road_functions::{ControlType, Intersection, RoadWay};
+use crate::road_parts::{Carriage, Designation, Lane, RoadEdge, E};
+use crate::units::{Direction, DrivingSide, Meters, Side, TrafficDirections};
+
+use abstio::MapName;
+use abstutil::Timer;
+use geom::Distance;
+use raw_map::{LaneSpec, LaneType, RawIntersection, RawMap, RawRoad};
+
+use enum_map::{enum_map, EnumMap};
+use itertools::Itertools;
+use std::collections::HashMap;
+
+/// ```
+/// use abstutil::Timer;
+/// use petgraph::dot::{Config, Dot};
+/// use streets::io::load_road_network;
+/// let mut timer = Timer::new("test osm2streets");
+/// let mut net = load_road_network(String::from("tests/src/aurora_sausage_link/input.osm"), &mut timer);
+/// println!("{}", net.to_dot());
+pub fn load_road_network(osm_path: String, timer: &mut Timer) -> RoadNetwork {
+    let driving_side = raw_map::DrivingSide::Right; // TODO
+    let clip = None;
+
+    let mut raw_map = convert_osm::convert(
+        osm_path.clone(),
+        MapName::new("zz", "osm2streets_test", &abstutil::basename(&osm_path)),
+        clip,
+        convert_osm::Options {
+            map_config: map_model::MapConfig {
+                driving_side,
+                bikes_can_use_bus_lanes: true,
+                inferred_sidewalks: true,
+                street_parking_spot_length: Distance::meters(8.0),
+                turn_on_red: true,
+            },
+            onstreet_parking: convert_osm::OnstreetParking::JustOSM,
+            public_offstreet_parking: convert_osm::PublicOffstreetParking::None,
+            private_offstreet_parking: convert_osm::PrivateOffstreetParking::FixedPerBldg(1),
+            include_railroads: true,
+            extra_buildings: None,
+            skip_local_roads: false,
+            filter_crosswalks: false,
+            gtfs_url: None,
+            elevation: false,
+        },
+        timer,
+    );
+
+    raw_map.run_all_simplifications(false, false, timer);
+
+    raw_map.into()
+}
+
+impl From<RawMap> for RoadNetwork {
+    fn from(map: RawMap) -> Self {
+        let mut net = RoadNetwork::new();
+        // Intersection ids from NodeIds
+        let is = HashMap::<_, _>::from_iter(map.intersections.iter().map(|(node_id, raw_int)| {
```

Comment:
Many ways to write this kind of thing. You could also do `let is: HashMap<_, _> = map.intersections.iter().map(...).collect()`, which I find a bit simpler. (I don't know if you can omit the type params like that)

Reply:
I do prefer invoking the common `collect` idiom. It looks like familiar boilerplate which you ought to ignore when you can :)